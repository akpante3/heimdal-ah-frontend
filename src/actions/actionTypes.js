import keyMirror from 'keymirror';

export const ACTIONS = keyMirror({
  OPEN_MODAL: null,
  CLOSE_MODAL: null,
  SET_CURRENT_USER: null,
  SET_CURRENT_USER_FAIL: null,
  LOGOUT_USER: null,
  RESET_PASSWORD_STATE: null,
  RESET_PASSWORD_SUCCESS: null,
  RESET_PASSWORD_FAILED: null,
  RESET_PASSWORD_ERROR: null,
  UPDATE_PASSWORD_SUCCESS: null,
  UPDATE_PASSWORD_FAILED: null,
  UPDATE_PASSWORD_ERROR: null,
});
