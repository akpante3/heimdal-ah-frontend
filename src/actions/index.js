import ResetPassword from './ResetPasswordActions';

export default ResetPassword;
