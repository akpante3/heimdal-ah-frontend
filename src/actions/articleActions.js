// import axios from 'axios';
import { ACTIONS } from './actionTypes';
import errorResponse from '../utils/errorResponse';
import axiosInstance from '../utils/axiosInstance';
import { toggleLoader } from './loaderActions';

export const reportArticle = (reportData = {}, toggleModal, flashMessage) => (dispatch) => {
  const { articleId } = reportData;
  axiosInstance
    .post(`/articles/${articleId}/reports`, reportData)
    .then(() => {
      dispatch(toggleLoader());
      flashMessage();
      setTimeout(() => {
        dispatch(() => {
          toggleModal();
          return {
            type: 'ACTIONS.TOGGLE_MODAL',
          };
        });
      }, 200);
    })
    .catch((errors = {}) => {
      console.log(errors);
      console.log(errors.response);
      const { response = {}, request } = errors;

      const data = {
        dispatch,
        request,
        response,
        errors,
      };
      dispatch(toggleLoader());
      return errorResponse(data);
    });
};
