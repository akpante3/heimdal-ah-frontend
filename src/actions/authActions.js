// import axios from 'axios';
import JWT from 'jsonwebtoken';
import { ACTIONS } from './actionTypes';
import { signupEndpoint } from '../apiEndpoints';
import errorResponse from '../utils/errorResponse';
import configureLocalStorage from '../utils/configureLocalStorage';
import axiosInstance, { setToken } from '../utils/axiosInstance';

export const setAuthUser = (token) => {
  const decodedToken = JWT.decode(token);
  const { userId, username, roleId } = decodedToken;

  configureLocalStorage.setAuthUser(token);
  // remember to set axios auth token here
  setToken(token);

  return {
    type: ACTIONS.SET_AUTH_USER,
    payload: {
      user: { userId, username, roleId },
    },
  };
};

export const signupUser = (signupData = {}, history) => (dispatch) => {
  axiosInstance
    .post(signupEndpoint, signupData)
    .then((response) => {
      const { token } = response.data;
      dispatch(setAuthUser(token));
      return history.push('/');
    })
    .catch((errors = {}) => {
      const { response = {}, request } = errors;

      const data = {
        dispatch,
        request,
        response,
        errors,
      };
      return errorResponse(data);
    });
};
