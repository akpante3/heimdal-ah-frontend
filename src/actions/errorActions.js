import { SET_ERRORS, CLEAR_ERRORS, REMOVE_AN_ERROR } from './actionTypes';

export const setErrors = (payload = {}) => ({
  type: SET_ERRORS,
  payload,
});

export const clearErrors = () => ({
  type: CLEAR_ERRORS,
});

export const removeAnError = errorName => ({
  type: REMOVE_AN_ERROR,
  errorName,
});
