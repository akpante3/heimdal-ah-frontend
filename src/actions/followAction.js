import axios from 'axios';
import { FOLLOW_USER, GET_ALL_FOLLOWERS, UNFOLLOW_USER } from './actionTypes';

// const token = window.localStorage.getItem('access-token');
// const paramsId = window.localStorage.getItem('followerId');
// const followedId = parseInt(paramsId, 10);
const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOjEsInVzZXJuYW1lIjoid2FsZSIsInJvbGVJZCI6MSwiaWF0IjoxNTQ4MzAzNDM5LCJleHAiOjE1NDgzODk4Mzl9.8UP-fhrGZXRUXn-xghL_gh89SBN_bO8bif1J3NM1NB0';

export const getAllFollowers = () => (dispatch) => {
  axios({
    method: 'get',
    url: 'https://heimdal-ah-staging.herokuapp.com/api/v1/users/followers',
    headers: { 'access-token': token },
  })
    .then((followData) => {
      console.log('axios=> get menu', followData);
      dispatch({
        type: GET_ALL_FOLLOWERS,
        payload: followData.data,
      });
    }).catch(err => console.log(err));
};

export const followUser = followedId => (dispatch) => {
  axios({
    method: 'post',
    url: `https://heimdal-ah-staging.herokuapp.com/api/v1/users/${followedId}/follow`,
    headers: { 'access-token': token },
  })
    .then((followData) => {
      console.log('axios=> get menu', followData);
      dispatch({
        type: FOLLOW_USER,
        // payload: followData.data,
        payload: { message: followData.data.message, followedId },
      });
    }).catch(err => console.log(err));
};

export const unfollowUser = followedId => (dispatch) => {
  axios({
    method: 'post',
    url: `https://heimdal-ah-staging.herokuapp.com/api/v1/users/${followedId}/unfollow`,
    headers: { 'access-token': token },
  })
    .then((followData) => {
      console.log('axios=> get menu', followData);
      dispatch({
        type: UNFOLLOW_USER,
        // payload: followData.data,
        payload: { message: followData.data.message, followedId },
      });
    }).catch(err => console.log(err));
};
