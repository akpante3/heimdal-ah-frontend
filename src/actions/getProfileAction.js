import axios from 'axios';
import { GET_ALL_PROFILE } from './actionTypes';

export const getAllProfile = () => (dispatch) => {
  axios
    .get('https://heimdal-ah-staging.herokuapp.com/api/v1/users')
    .then((profileData) => {
      console.log('axios=> get menu', profileData);
      dispatch({
        type: GET_ALL_PROFILE,
        payload: profileData.data,
      });
    }).catch(err => console.log(err));
};
