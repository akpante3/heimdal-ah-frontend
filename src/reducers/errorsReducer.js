import { SET_ERRORS, REMOVE_AN_ERROR, CLEAR_ERRORS } from '../actions/actionTypes';

const errorsReducer = (state = {}, action) => {
  switch (action.type) {
    case SET_ERRORS:
      return { ...action.payload };

    case REMOVE_AN_ERROR:
      return { ...state, [action.errorName]: '' };

    case CLEAR_ERRORS:
      return {};

    default:
      return state;
  }
};

export default errorsReducer;
