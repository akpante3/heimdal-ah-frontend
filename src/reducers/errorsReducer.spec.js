/* eslint-disable no-undef */
import { ACTIONS } from '../actions/actionTypes';
import { removeAnError, clearErrors } from '../actions/errorActions';

// Reducer to be tested
import errorsReducer from './errorsReducer';

describe('authReducer', () => {
  it('should return the default state if no action is specified', () => {
    const initialState = {};
    const action = { type: '' };
    expect(errorsReducer(initialState, action)).toMatchSnapshot({});
    expect(errorsReducer(initialState, action)).toEqual({});
  });

  it('should update the state and set errors when type is SET_ERRORS and action is specified', () => {
    const initialState = {};
    const action = { type: ACTIONS.SET_ERRORS, payload: { username: ['username is required'] } };
    expect(errorsReducer(initialState, action)).toMatchSnapshot({});
    expect(errorsReducer(initialState, action)).toEqual({ username: ['username is required'] });
  });

  it('should update the state and remove an error when type is REMOVE_AN_ERRORS and errorName is specified in the action', () => {
    const initialState = {};
    const action = removeAnError('username');
    expect(errorsReducer(initialState, action)).toMatchSnapshot({});
    expect(errorsReducer(initialState, action)).toEqual({ ...initialState, username: '' });
  });

  it('should clear all the errors in state when the type is REMOVE_AN_ERRORS', () => {
    const initialState = {};
    const action = clearErrors();
    expect(errorsReducer(initialState, action)).toMatchSnapshot({});
    expect(errorsReducer(initialState, action)).toEqual({});
  });
});
