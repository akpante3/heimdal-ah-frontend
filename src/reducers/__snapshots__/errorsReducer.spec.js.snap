// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authReducer should clear all the errors in state when the type is REMOVE_AN_ERRORS 1`] = `Object {}`;

exports[`authReducer should return the default state if no action is specified 1`] = `Object {}`;

exports[`authReducer should update the state and remove an error when type is REMOVE_AN_ERRORS and errorName is specified in the action 1`] = `
Object {
  "username": "",
}
`;

exports[`authReducer should update the state and set errors when type is SET_ERRORS and action is specified 1`] = `
Object {
  "username": Array [
    "username is required",
  ],
}
`;
